common/src/main/java/org/apache/comet/vector/CometPlainVector.java

```diff
@@ -113,7 +113,11 @@ public UTF8String getUTF8String(int rowId) {
       byte[] result = new byte[length];
       Platform.copyMemory(
           null, valueBufferAddress + offset, result, Platform.BYTE_ARRAY_OFFSET, length);
-      return UTF8String.fromString(convertToUuid(result).toString());
+      if (length == 16) {
+        return UTF8String.fromString(convertToUuid(result).toString());
+      } else {
+        return UTF8String.fromBytes(result);
+      }
```

Comment:
In TPCDS queries, this encounters non uuid (not 16 bytes) FLBA (fixed length byte array) to be accessed as string. Actually, I am not sure about the uuid mapping (FLBA -> string) statement here:

> // Iceberg maps UUID to StringType.
> // The data type here must be UUID because the only FLBA -> String mapping we have is UUID.

So looks like this is not correct because it actually has non uuid FLBA -> string mapping. 🤔 
This may lead to possible issue if the FLBA is happened to be 16 bytes but it is NOT uuid but normal string.

But without this change, the query cannot be passed.

Is there any metadata info we can get to know if it is an uuid column or not?

cc @huaxingao

Reply:
Hm? Isn't the current `else` a general handling of FLBA (bytes) to string?

Reply:
What I mean is we remove the logic of UUID for now

```scala
  @Override
  public UTF8String getUTF8String(int rowId) {
    if (!isBaseFixedWidthVector) {
      BaseVariableWidthVector varWidthVector = (BaseVariableWidthVector) valueVector;
      long offsetBufferAddress = varWidthVector.getOffsetBuffer().memoryAddress();
      int offset = Platform.getInt(null, offsetBufferAddress + rowId * 4L);
      int length = Platform.getInt(null, offsetBufferAddress + (rowId + 1L) * 4L) - offset;
      return UTF8String.fromAddress(null, valueBufferAddress + offset, length);
    } else {
      BaseFixedWidthVector fixedWidthVector = (BaseFixedWidthVector) valueVector;
      int length = fixedWidthVector.getTypeWidth();
      int offset = rowId * length;
      byte[] result = new byte[length];
      Platform.copyMemory(
          null, valueBufferAddress + offset, result, Platform.BYTE_ARRAY_OFFSET, length);
      return UTF8String.fromBytes(result);
```

Reply:
It will drop the Iceberg UUID stuff, is it okay?

Reply:
Yea, since Iceberg is not supported yet. We should think about how to properly support UUID later. Currently it just check whether length is 16 and treat it as UUID if so, which doesn't seem correct.

Reply:
Yea, okay.

Reply:
cc @snmvaughan for uuid stuff.